dashboard/app/controllers/reports_controller.rb

```diff
@@ -56,6 +56,25 @@ def all_usage
     render 'usage', formats: [:html]
   end
 
+  def search_for_teachers
+    authorize! :read, :reports
+    fields = ['id', 'name', 'email', 'address', 'num_students']
+    if (params[:emailFilter])
+      email_filter = '%' + params[:emailFilter] + '%'
+    else
+      email_filter = '%%'
+    end
+    if (params[:addressFilter])
+      address_filter = '%' + params[:addressFilter] + '%'
+    else
+      address_filter = '%%'
+    end
+
+    teachers = User.limit(100).where(user_type: 'teacher').where("email LIKE ?", email_filter).where("full_address LIKE ?", address_filter).joins(:followers).group('followers.user_id').pluck(:id, :name, :email, :full_address, 'COUNT(followers.id) AS num_students')
```

Comment:
would it be useful to allow pagination of results > 100? or at least some indication that there are more than 100 results.